components/external-link/index.js

```diff
@@ -8,33 +8,76 @@ import { compact, uniq } from 'lodash';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Dashicon from '../dashicon';
+import Button from '../button';
 import './style.scss';
 
-function ExternalLink( { href, children, className, rel = '', ...additionalProps } ) {
-	rel = uniq( compact( [
-		...rel.split( ' ' ),
-		'external',
-		'noreferrer',
-		'noopener',
-	] ) ).join( ' ' );
-	const classes = classnames( 'components-external-link', className );
-	return (
-		<a { ...additionalProps } className={ classes } href={ href } target="_blank" rel={ rel }>
-			{ children }
+class ExternalLink extends Component {
+	getRel() {
+		const { rel = '' } = this.props;
+
+		// Allow to omit the `rel` attribute passing a `null` value.
+		return rel === null ? rel : uniq( compact( [
+			...rel.split( ' ' ),
+			'external',
+			'noreferrer',
+			'noopener',
+		] ) ).join( ' ' );
+	}
+
+	renderOpensInNewTabText() {
+		const { opensInNewTabText = __(
+			/* translators: accessibility text */
+			'(opens in a new tab)'
+		) } = this.props;
+
+		return (
 			<span className="screen-reader-text">
 				{
-					/* translators: accessibility text */
-					__( '(opens in a new window)' )
+					// We need a space to separate this from previous text.
+					' ' + opensInNewTabText
 				}
 			</span>
-			<Dashicon icon="external" />
-		</a>
-	);
+		);
+	}
+
+	render() {
+		const {
+			href,
+			target,
+			icon = true,
+			children,
+			className,
+			...additionalProps
+		} = this.props;
+
+		const { disabled, isPrimary, isLarge, isSmall } = additionalProps;
+		const targetToUse = target || '_blank';
+		const isDisabled = disabled || ! href;
```

Comment:
What's the use case for rendering this without `href` ?

Reply:
> if no href props is passed, it's rendered as a disabled button